samples/bluetooth/hci_uart/sample.yaml

```diff
@@ -6,12 +6,20 @@ tests:
     harness: bluetooth
     platform_allow:
       - nrf52dk/nrf52832
+      - nrf52840dk/nrf52840
+      - nrf21540dk/nrf52840
```

Comment:
This whole list doesn't make much sense to me. We should allow any platform that has Bluetooth support and a UART. With a list like this, wanting to build/run this with twister on any other platform requires passing an override command line flag (`--force-platform`), which isn't very friendly.

Reply:
i believe `integration_platforms` is a subset of `platform_allow` which ensures these samples are buildable for these boards, and ensured in PR to pass the building.

Other boards that support being a Controller only that can be build should be included in the future here.